### PR TITLE
API pull's head/base have correct permission

### LINKED ISSUE #17181: [API] `/repos/{owner}/{repo}/pulls/{index}` returns `false` for all permissions of the base/head branch repo
### Gitea Version

(almost) latest main

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Built from source, from cli

### Database

_No response_

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Description

If I get a PR via the API, it returns the pull's head/base branch. The repos that are included in these information also return the permissions, but all values of these permissions (`admin`, `push`, `pull`) are `false`.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by pricly-yellow**
Closes #17181

* for all pull requests API return permissions of caller
* for all webhook return empty permissions

### CODE REVIEW

File: modules/convert/pull.go [hunk 1]
```diff
@@ -41,6 +41,12 @@ func ToAPIPullRequest(pr *models.PullRequest) *api.PullRequest {
 		return nil
 	}
 
+	perm, err := models.GetUserRepoPermission(pr.BaseRepo, user)
+	if err == nil {
```
Comment (delvh) [inline modules/convert/pull.go]:
```suggestion
	if err != nil {
```

File: modules/convert/pull.go [hunk 2]
```diff
@@ -114,8 +120,14 @@ func ToAPIPullRequest(pr *models.PullRequest) *api.PullRequest {
 	}
 
 	if pr.HeadRepo != nil && pr.Flow == models.PullRequestFlowGithub {
+		perm, err := models.GetUserRepoPermission(pr.HeadRepo, user)
+		if err == nil {
```
Comment (delvh) [inline modules/convert/pull.go]:
```suggestion
		if err != nil {
```

**Review by delvh — APPROVED** [review]

File: modules/convert/pull.go [hunk 3]
```diff
@@ -17,7 +17,7 @@ import (
 // ToAPIPullRequest assumes following fields have been assigned with valid values:
 // Required - Issue
 // Optional - Merger
-func ToAPIPullRequest(pr *models.PullRequest) *api.PullRequest {
+func ToAPIPullRequest(pr *models.PullRequest, user *models.User) *api.PullRequest {
```
Comment (6543) [inline modules/convert/pull.go]: Can we rename 'user' to 'doer' to indicate value source

**Review by noerw — APPROVED** [review]

File: modules/convert/pull.go (hunk 3 above)
Comment (pricly-yellow) [reply modules/convert/pull.go]: Good idea, pull it

**Review by 6543 — APPROVED** [review]

**Pull request merged**